src/core/net/ip6.cpp

```diff
@@ -358,6 +422,7 @@ otError Ip6::SendDatagram(Message &aMessage, MessageInfo &aMessageInfo, IpProto
     const NetifUnicastAddress *source;
 
     header.Init();
+    header.SetDscp(PriorityToDscp(aMessage.GetPriority()));
```

Comment:
Just a question.. While i am okay with this change, Thread 1.1 specification says that Traffic Class field shall be set to 0 on transmission and ignored on reception - although i 
guess its not a violation. The other downside of this change is that each message with high priority e.g. MLE frame will be increased by 1 byte due to 6LoWPAN compresion. Is that okay?

Reply:
The priority of MLE and Thread Mgmt messages are set to kPriorityNet by default. The priority kPriorityNet is mapped to kDscpCs0 in the function `PriorityToDscp()`, the frame won't increase 1 byte due to 6LoWPAN compression. Priority kPriorityNet is identified by the destination UDP port, please refer to the function `GetFramePriority()` in mesh_forwarder.cpp. For user application traffic, in order to get better performance, I think it is worth adding a byte.